azurelinuxagent/common/protocol/wire.py

```diff
@@ -1194,6 +1214,8 @@ def get_artifacts_profile(self):
 
             try:
                 profile = self.send_request_using_appropriate_channel(direct_func, host_func)
+                if profile is None:
+                    raise Exception("Failed to fetch artifacts profile from blob {0}".format(blob))
```

Comment:
This is a redundant raise, this will be caught in the next line and then eventually None would be returned. Why not just handle it like that to begin with (with proper logging ofcourse)

Reply:
I intentionally put that raise so I can re-use the code that logs the warning and returns None, but I realized the error message in that case would look weird: "Exception retrieving artifacts profile: Failed to fetch artifacts profile from blob [blob]". I decided to go with your suggestion, so this is updated now.